### ISSUE TITLE
Improvements and Bugs on Review Applications page

### ISSUE THREAD

**Issue by crisgarner** [privacy-scaling-explorations/maci-rpgf#93]
- [x] After a project has been approved it takes a while for the status to be updated (even though on the other page, it appears accepted. 
- [ ] There is no way to reject projects.
- [x] Filter / Search projects for the coordinators.
- [x] Once an application is approved it takes a lot to be displayed as approved, leaving to some UI bugs.

Screenshots are visible on: https://www.notion.so/pse-team/RPGF-feedback-1476b27526574f8aa8ae677988b7f135

**Comment by aguzmant103** [comment]
Could have been solved in Easy-RetroPGF like in commit 
Careful with ehterjs version
https://github.com/gitcoinco/easy-retro-pgf/commit/2a5cd405eb3b99e0724c8b6ede089830ad6df01a

**Comment by aguzmant103** [comment]
- [ ] There is no way to reject projects once they have been approved

**Comment by ctrlc03** [comment]
@aguzmant103 @crisgarner we could probably have a button to revoke the attestation once approved, but actually rejecting an application might be difficult (for that we probably would need to create another schema "rejected applications" and map an application to this new eas schema and do the filtering while pulling the data)

**Comment by crisgarner** [comment]
We could update the query to not show it once revoked, the main issue is that the applications always stay on the admin page. We can use the same schema as the one we use is revokable

https://github.com/privacy-scaling-explorations/maci-rpgf/blob/86228a02fcc5b230a0bf3fb60abe3ac7d8016725/src/utils/fetchAttestations.ts#L50

**Comment by ctrlc03** [comment]
> We could update the query to not show it once revoked, the main issue is that the applications always stay on the admin page. We can use the same schema as the one we use is revokable
> 
> https://github.com/privacy-scaling-explorations/maci-rpgf/blob/86228a02fcc5b230a0bf3fb60abe3ac7d8016725/src/utils/fetchAttestations.ts#L50

can we revoke the one created by the user when applying as a way of rejecting?

**Comment by crisgarner** [comment]
Now that you mention it, not sure if we can revoke their attestation as they created it. Currently, the flow is: 
1. User creates an application for the round, [for example](https://optimism.easscan.org/attestation/view/0xf3f00737db5bf56baa75a23b7cbfa4e3c4355c330ccf43c0e5d2623484291bc2)
2. The coordinator/admin creates an attestation and references the previous one. [ example](https://optimism.easscan.org/attestation/view/0x9396cea4418db77f9de53ce923d3e90fe409d1f439a6b7851c862e1a1330f722)

We kind of assume that any attestation done is an approval, a quick fix would be to create another approval schema with the status (approved or rejected).

**Comment by ctrlc03** [comment]
> Now that you mention it, not sure if we can revoke their attestation as they created it. Currently, the flow is:
> 
> 1. User creates an application for the round, [for example](https://optimism.easscan.org/attestation/view/0xf3f00737db5bf56baa75a23b7cbfa4e3c4355c330ccf43c0e5d2623484291bc2)
> 2. The coordinator/admin creates an attestation and references the previous one. [ example](https://optimism.easscan.org/attestation/view/0x9396cea4418db77f9de53ce923d3e90fe409d1f439a6b7851c862e1a1330f722)
> 
> We kind of assume that any attestation done is an approval, a quick fix would be to create another approval schema and add the status field to save if approved or rejected.

as we cannot revoke it cause we didn't create it, feels like we can't even change the status, correct me if I'm wrong here?!

**Comment by ctrlc03** [comment]
> Now that you mention it, not sure if we can revoke their attestation as they created it. Currently, the flow is:
> 
> 1. User creates an application for the round, [for example](https://optimism.easscan.org/attestation/view/0xf3f00737db5bf56baa75a23b7cbfa4e3c4355c330ccf43c0e5d2623484291bc2)
> 2. The coordinator/admin creates an attestation and references the previous one. [ example](https://optimism.easscan.org/attestation/view/0x9396cea4418db77f9de53ce923d3e90fe409d1f439a6b7851c862e1a1330f722)
> 
> We kind of assume that any attestation done is an approval, a quick fix would be to create another approval schema and add the status field to save if approved or rejected.

Ok right so we basically create one same way as if we are approving it but set it to rejected, and then can filter later based on that field. that works

**Comment by crisgarner** [comment]
> > Now that you mention it, not sure if we can revoke their attestation as they created it. Currently, the flow is:
> > 
> > 1. User creates an application for the round, [for example](https://optimism.easscan.org/attestation/view/0xf3f00737db5bf56baa75a23b7cbfa4e3c4355c330ccf43c0e5d2623484291bc2)
> > 2. The coordinator/admin creates an attestation and references the previous one. [ example](https://optimism.easscan.org/attestation/view/0x9396cea4418db77f9de53ce923d3e90fe409d1f439a6b7851c862e1a1330f722)
> > 
> > We kind of assume that any attestation done is an approval, a quick fix would be to create another approval schema and add the status field to save if approved or rejected.
> 
> Ok right so we basically create one same way as if we are approving it but set it to rejected, and then can filter later based on that field. that works

yeah exactly, we wouldn't need to revoke anymore, just create an attestation and set the status to false, and when approving set it to true.

**Comment by ctrlc03** [comment]
> > > Now that you mention it, not sure if we can revoke their attestation as they created it. Currently, the flow is:
> > > 
> > > 1. User creates an application for the round, [for example](https://optimism.easscan.org/attestation/view/0xf3f00737db5bf56baa75a23b7cbfa4e3c4355c330ccf43c0e5d2623484291bc2)
> > > 2. The coordinator/admin creates an attestation and references the previous one. [ example](https://optimism.easscan.org/attestation/view/0x9396cea4418db77f9de53ce923d3e90fe409d1f439a6b7851c862e1a1330f722)
> > > 
> > > We kind of assume that any attestation done is an approval, a quick fix would be to create another approval schema and add the status field to save if approved or rejected.
> > 
> > 
> > Ok right so we basically create one same way as if we are approving it but set it to rejected, and then can filter later based on that field. that works
> 
> yeah exactly, we wouldn't need to revoke anymore, just create an attestation and set the status to false, and when approving set it to true.

will open as a separate issue

**Comment by kittybest** [comment]
split the first issue to #213

**Comment by kittybest** [comment]
the second issues is re-opened here: #193 
the first and fourth issues are re-opened here: #213, and already implemented.
the third issue is closed by #82, if there's any more specific description for this problem, please just open another new issue.
